### ISSUE TITLE
Failure writing output to destination

### ISSUE THREAD

**Issue by hajsf** [Sharktheone/arch-mojo#1]
Hi, I'm at Manjaro, and got the mentioned error:
```bach
 $ python --version 
Python 3.11.5
 $ curl -sSL https://raw.githubusercontent.com/Sharktheone/arch-mojo/main/install.py -o /tmp/arch-mojo/install.py && python /tmp/arch-mojo/install.py
```

**Comment by Sharktheone** [comment]
I think this is because i forgot to add `mkdir -p /tmp/arch-mojo`in front of the curl command. I can update this in the script later. For now, just execute the mkdir command and rerun the install command.

**Comment by hajsf** [comment]
> mkdir -p /tmp/arch-mojo

Thanks I got the below:
```bach
$ mkdir -p /tmp/arch-mojo 
$ curl -sSL https://raw.githubusercontent.com/Sharktheone/arch-mojo/main/install.py -o /tmp/arch-mojo/install.py && python /tmp/arch-mojo/install.py

Do you have modular already installed? (y/n): n
Do you want to install the libraries globally (for all users)? (y/n): y
Do you want to automatically use a venv when running modular install/update? (y/n): y
Where do you want to create the venv?: ~/.virtualenvs
Please enter your Modular auth token. You can also type 'manual' to run modular manually when requested: mut_a2xxxxxxxxxxxxxxxxxxxxxxxxx

# Found release for https://packages.modular.com/mojo @ 0.3.0, installing to /home/hasany/.modular/pkg/packages.modular.com_mojo
# Downloads complete, setting configs...
# Configs complete, running post-install hooks...
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try 'pacman -S
    python-xyz', where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Arch-packaged Python package,
    create a virtual environment using 'python -m venv path/to/venv'.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip.
    
    If you wish to install a non-Arch packaged Python application,
    it may be easiest to use 'pipx install xyz', which will manage a
    virtual environment for you. Make sure you have python-pipx
    installed via pacman.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
modular: error: failed to run python: 
Traceback (most recent call last):
  File "/tmp/arch-mojo/install.py", line 131, in <module>
    ("~/.local/lib/mojo" not in os.environ["LD_LIBRARY_PATH"]
                                ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
  File "<frozen os>", line 679, in __getitem__
KeyError: 'LD_LIBRARY_PATH'
```

What shall I do?

**Comment by Sharktheone** [comment]
This is the python externally managed thing, the script should avoid that by activating a venv, but i think, the os.system() function in python does not save "session". I have changed this in the script. You can try it again now.

**Comment by hajsf** [comment]
Thanks, now sure if the blew that I got is meaning modular was already installed in the previouse step or now, but I run the command and answered that I do not have modular installed, and got:
```bach
:: Processing package changes...
(1/1) reinstalling modular                                                                                                      [#############################################################################] 100%
ldconfig: /usr/lib/libmenu.so.6 is not a symbolic link

ldconfig: /usr/lib/libform.so.6 is not a symbolic link

ldconfig: /usr/lib/libpanel.so.6 is not a symbolic link

ldconfig: /usr/lib/libedit.so.2 is not a symbolic link

ldconfig: /usr/lib/libncurses.so.6 is not a symbolic link

:: Running post-transaction hooks...
(1/2) Arming ConditionNeedsUpdate...
(2/2) Refreshing PackageKit...
x - debian-binary
x - control.tar.xz
x - data.tar.xz
x - debian-binary
x - control.tar.xz
x - data.tar.xz
# Found release for https://packages.modular.com/mojo @ 0.3.0, installing to /home/hasany/.modular/pkg/packages.modular.com_mojo
modular: error: failure detected - please run `modular clean` and try again
==========================
Failure Information:
        - Creating symlink for /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/liblldb.so.18.0.0git -> /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/liblldb.so failed: File exists
        - Creating symlink for /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libMojoJupyter.so.18git -> /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libMojoJupyter.so failed: File exists
        - Creating symlink for /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/liblldb.so.18.0.0git -> /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/liblldb.so.18git failed: File exists
        - Creating symlink for /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libKGENCompilerRTShared.so.18git -> /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libKGENCompilerRTShared.so failed: File exists
        - Creating symlink for /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libMojoLLDB.so.18git -> /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libMojoLLDB.so failed: File exists
        - Creating symlink for /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libLLCLRuntimeGlobals.so.18git -> /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libLLCLRuntimeGlobals.so failed: File exists
        - http error: failed to reach URL https://cas.modular.com/0/BJQxbJ7r1e1hCtY-5drp4KPW-MeoBkBY3LmSRHBhMek with cURL error Failure when receiving data from the peer - Looking for BJQxbJ7r1e1hCtY-5drp4KPW-MeoBkBY3LmSRHBhMek
        - Creating symlink for /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libMSupportGlobals.so.18git -> /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libMSupportGlobals.so failed: File exists
==========================
```

Then I got:
```
$ modular clean
zsh: correct 'modular' to '.modular' [nyae]? y
zsh: command not found: .modular
```
Then I tried again
```
$ modular clean 
$ modulat install mojo
# Found release for https://packages.modular.com/mojo @ 0.3.0, installing to /home/hasany/.modular/pkg/packages.modular.com_mojo
```
It looks to be taking long time, I'm waiting since a while and nothing happened.

Then I got:
```bach
modular install mojo
# Found release for https://packages.modular.com/mojo @ 0.3.0, installing to /home/hasany/.modular/pkg/packages.modular.com_mojo
# Downloads complete, setting configs...
# Configs complete, running post-install hooks...
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try 'pacman -S
    python-xyz', where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Arch-packaged Python package,
    create a virtual environment using 'python -m venv path/to/venv'.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip.
    
    If you wish to install a non-Arch packaged Python application,
    it may be easiest to use 'pipx install xyz', which will manage a
    virtual environment for you. Make sure you have python-pipx
    installed via pacman.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
modular: error: failed to run python: 
```

What do you think, what is the issue, and what shall I do now? Thanks.

**Comment by Sharktheone** [comment]
So when you run modular, you need to activate a venv or run it with a environment option. the fistst log you sent shows a successful installation, the installation "failed" because it is already installed as you can se at `failed: File exists` when you can run `mojo` or ` ~/.modular/pkg/packages.modular.com_mojo/bin/mojo` you have sucessfully installed mojo 🔥. You can also use the option to create a venv automatically when you run modular, but after this you will need to reopen your shell or reactivate your shell-rc file.

**Comment by hajsf** [comment]
Thanks, but I'm not getting mojo installed:
```bach
$ modular install mojo                                                                                                                                                                             1 
# Found release for https://packages.modular.com/mojo @ 0.3.0, installing to /home/hasany/.modular/pkg/packages.modular.com_mojo
modular: error: failure detected - please run `modular clean` and try again
==========================
Failure Information:
        - Creating symlink for /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/liblldb.so.18.0.0git -> /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/liblldb.so failed: File exists
        - Creating symlink for /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libMojoJupyter.so.18git -> /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libMojoJupyter.so failed: File exists
        - Creating symlink for /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/liblldb.so.18.0.0git -> /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/liblldb.so.18git failed: File exists
        - Creating symlink for /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libKGENCompilerRTShared.so.18git -> /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libKGENCompilerRTShared.so failed: File exists
        - Creating symlink for /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libMojoLLDB.so.18git -> /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libMojoLLDB.so failed: File exists
        - Creating symlink for /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libLLCLRuntimeGlobals.so.18git -> /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libLLCLRuntimeGlobals.so failed: File exists
        - Creating symlink for /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libMSupportGlobals.so.18git -> /home/hasany/.modular/pkg/packages.modular.com_mojo/lib/libMSupportGlobals.so failed: File exists
==========================

$ mojo                                                                                                                                                                                   
zsh: command not found: mojo
```

**Comment by Sharktheone** [comment]
What happens, when you restart your shell or use `~/.modular/pkg/packages.modular.com_mojo/bin/mojo` ?

**Comment by hajsf** [comment]
> ~/.modular/pkg/packages.modular.com_mojo/bin/mojo

After restarting the system, I got:
```bach
$ mojo
zsh: command not found: mojo

$ ~/.modular/pkg/packages.modular.com_mojo/bin/mojo
Welcome to Mojo! 🔥
Expressions are delimited by a blank line.
Type `:quit` to exit the REPL and `:mojo help` for further assistance.

  1> 
```

Then I installed mojo extenstion for Github, and got:
```bach
The Mojo🔥 development environment was not found. If the Mojo SDK is installed, please set the MODULAR_HOME environment variable to the appropriate path, or set the `mojo.modularHomePath` configuration. If you do not have it installed, would you like to install it?
```

Then I selected `settings:`
```bach
Mojo: Modular Home Path
The path to the MODULAR_HOME containing the Mojo SDK installation.
~/.modular/pkg/packages.modular.com_mojo/bin/
```
I opened the `~/.bashrc` and found the below 2 lines had been added at the end:
```bach
# added by arch-mojo script
export PATH=$PATH:~/.modular/pkg/packages.modular.com_mojo/bin/
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:~/.local/lib/mojo
```

I opened the file `/etc/environment` and set `  MODULAR_HOME =~/.modular/pkg/packages.modular.com_mojo/bin`.

And now I have the below:

1. At the system shell:
 - $ mojo --versi
 - zsh: command not found: mojo
 - $ ~/.modular/pkg/packages.modular.com_mojo/bin/mojo --version
 - mojo 0.3.0 (f64f9601)
 
2. At VS code:
- The extinsion is not yet seeing the MODULAR_HOME even after restarting the vscode
- The terminal looks to be seeing the MODULAR_HOME and I got:
 - $ mojo --version
    mojo 0.3.0 (f64f9601)
 

Did I made something wrong? and how can I make:
1. MODULAR_HOME seen by the system shell
2. MODULAR_HOME seen by the VSCode extension


**UPDATE**
As I'm using `zsh` I found that I have to update the `~/.zshrc` file, so I did the below:

1. Open the `zsh` profile file with a text editor:

```bash
$ code ~/.zshrc
```

2. Add the following line at the end of the file:

```bash
export PATH="$HOME/.modular/pkg/packages.modular.com_mojo/bin:$PATH"
```

3. Save and close the file.

4. To make the changes take effect, either by restarting the shell or sourcing the profile file with the following command:

```bash
$ source ~/.zshrc
```

Run the `mojo` command directly from any directory in the shell.:

```bash
$ mojo --version
mojo 0.3.0 (f64f9601)
```

For the `MODULAR_HOME` variable, you can follow similar steps:

1. Open the `zsh` profile file with a text editor:

```bash
$ code ~/.zshrc
```

2. Add the following line at the end of the file:

```bash
export MODULAR_HOME="$HOME/.modular"
```

3. Save and close the file.

4. To make the changes take effect, either restart your shell or source the profile file with the following command:

```bash
$ source ~/.zshrc
```

Now, the system will recognize `MODULAR_HOME` as an environment variable pointing to the modular home directory. Can be verified by running:

```bash
$ echo $MODULAR_HOME
/home/hasany/.modular
```

Now my `~/.zshrc` file is:
```bach
# Use powerline
USE_POWERLINE="true"
# Source manjaro-zsh-configuration
if [[ -e /usr/share/zsh/manjaro-zsh-config ]]; then
  source /usr/share/zsh/manjaro-zsh-config
fi
# Use manjaro zsh prompt
if [[ -e /usr/share/zsh/manjaro-zsh-prompt ]]; then
  source /usr/share/zsh/manjaro-zsh-prompt
fi

export PATH="$HOME/.modular/pkg/packages.modular.com_mojo/bin:$PATH"
export MODULAR_HOME="$HOME/.modular"
```

Now it looks `mojo` is fine, but VSCode extension still not seeing the `MODULAR_HOME` though I defined it in the extension setting as  `$HOME/.modular`!

**UPDATE 2**
I defined the `$HOME/.modular` at mojo extension for VSCode as: `/home/hasany/.modular`
```bach
Mojo: Modular Home Path
The path to the MODULAR_HOME containing the Mojo SDK installation.
/home/hasany/.modular
```

And it is working fine now.

Not sure if this could help you fixing something in the code, or at least hope other users getting benifit from my case.

Thanks a lot.

**UPDATE 3**
I tried to run a simple code as:
```rs
let x =10
fn main():
    print(x)
```

But got the below error `/.modular/pkg/packages.modular.com_mojo/bin/crashdb: No such file or directory`:
```bach
$ mojo main.mojo
[2491:2491:20230924,133050.068000:ERROR filesystem_posix.cc:63] mkdir /home/hasany/.modular/pkg/packages.modular.com_mojo/bin:/home/hasany/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/hasany/.modular/pkg/packages.modular.com_mojo/bin/crashdb: No such file or directory (2)
[2493:2493:20230924,133050.070315:ERROR filesystem_posix.cc:63] mkdir /home/hasany/.modular/pkg/packages.modular.com_mojo/bin:/home/hasany/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/hasany/.modular/pkg/packages.modular.com_mojo/bin/crashdb: No such file or directory (2)
[2491:2491:20230924,133050.070609:ERROR socket.cc:120] recvmsg: Connection reset by peer (104)
Failed to initialize Crashpad.  Crash reporting will not be available.  Cause: crashpad failed to start handler
/home/hasany/Documents/mogo-tutorial/main.mojo:1:1: error: unable to locate module 'builtin'
let x =10
^
/home/hasany/Documents/mogo-tutorial/main.mojo:1:8: error: could not find builtin 'IntLiteral' type
let x =10
       ^
/home/hasany/Documents/mogo-tutorial/main.mojo:3:5: error: use of unknown declaration 'print', 'fn' declarations require explicit variable declarations
    print(x)
    ^~~~~
mojo: error: failed to parse the provided Mojo
```

**Comment by Sharktheone** [comment]
Thanks for the detailed post. I can look into it, updating the installation script, reduce manual intervention. For the last error you sent in update 3, I think, this is not a problem with mojo or the installation script.

What is also weird, is that it prints out your `$PATH` env variable and tries to mkdir's it. This does not make sense. But i also think, the crashpad is not soo important for now, since it is probably only error reporting to modular.

It mentions something with unknown declaration of `print`. It seems like somehow builtin functions and types are missing in your installation. Can you maybe try reinstalling mojo, if you haven't done that?

**Comment by hajsf** [comment]
> Can you maybe try reinstalling mojo, if you haven't done that?

I removed the .modular folder, and reinstalled everything, I noticed the installer is telling me about adding the paths and environment variable to the zhrc file.
But I still getting the same error at about builtin function.

**Comment by Sharktheone** [comment]
Hmm, i have seen in your 2nd comment, you have the libraries installed globally. Can you try, installing it in a local folder? You also need to delete the libraries in `/usr/lib/` and `/lib` .  But it is important, you only delete the libraries with the exact version extension, those beeing:
- `libncurses.so.6` and `libncurses.so.6.4`
- `libform.so.6` and `libform.so.6.4`
- `libpanel.so.6`and `libpanel.so.6.4`
- `libedit.so.2` and `libedit.so.2.0.70`

**Comment by mdmrk** [comment]
Starting from clean install (manually deleted all modular/mojo related folders) it fails somehow. Maybe somebody is encountering the same issue

```sh
$ mkdir -p /tmp/arch-mojo && curl -sSL https://raw.githubusercontent.com/Sharktheone/arch-mojo/main/install.py -o /tmp/arch-mojo/install.py && python /tmp/arch-mojo/install.py
Do you have modular already installed? (y/n): n
Do you want to install the libraries globally (for all users)? (y/n): n
Do you want to automatically use a venv when running modular install/update? (y/n): y
Where do you want to create the venv?: zsh

(...)

* `/home/mrk/.modular/pkg/packages.modular.com_mojo/bin/mojo`...
TEST: `mojo --help`... OK
TEST: `mojo run --help`... OK
TEST: `mojo build test_mandelbrot.mojo`... OK
TEST: `mojo build test_python.mojo`... OK
TEST: `mojo demangle`... OK
TEST: `mojo format`... OK
TEST: `mojo package`... OK
TEST: `mojo test_mandelbrot.mojo`... OK
TEST: `mojo test_python.mojo`... OK
== stdout ==

== stderr ==
/home/mrk/.modular/pkg/packages.modular.com_mojo/bin/lldb: error while loading shared libraries: libpanel.so.6: cannot open shared object file: No such file or directory

(...)
```

**Comment by Sharktheone** [comment]
Another thing that could be a problem, is manjaro. Maybe manjaro behaves a bit differently from arch. I can test this later with manjaro. If i have the same problems, it might be easier to resolve these.

**Comment by Sharktheone** [comment]
@mdmrk i have created a new issue from this, because i think, this is not related to this issue.

**Comment by Sharktheone** [comment]
> I can test this later with manjaro.

I've tested it on manjaro and it works like on arch for me. I also have seen, a post from you on the Modular Discord, so i guess you jave get it running?

**Comment by hajsf** [comment]
> I also have seen, a post from you on the Modular Discord

I installed Ubuntu to be able to start and have feeling if Mojo, but we'll return to Manjaro soon.

Note: With Ubuntu it was installed properly only after activating the python wirking environment, which is not mentioed in the setup page

**Comment by XilinJia** [comment]
I confirm the installation succeeded on Manjaro.

In a Python 3.10 venv:

`➜  Python python <(curl -sSL https://raw.githubusercontent.com/Sharktheone/arch-mojo/main/install.py)
`

It asked for auth code, which I didn't have but got it by creating an account at modular's site, input the code, then it started installing quite a lot of packages, then succeeded.

```
➜  Mojo mojo --version
mojo 0.3.1 (a3eed7c8)
➜  Mojo mojo
Welcome to Mojo! 🔥
Expressions are delimited by a blank line.
Type `:quit` to exit the REPL and `:mojo help` for further assistance.
```

**Comment by Sharktheone** [comment]
Thanks for your Message. I also have only really tested the updated scripts on Manjaro, since I only changed the way of installing and I already installed it on my both Arch systems.

**Comment by hajsf** [comment]
I got now a fresh install of `Archcraft` linux, and got `Mojo` installed, but forced to do `modular clean` before installing `mojo`

```bach
~ python --version
Python 3.11.5
~ python <(curl -sSL https://raw.githubusercontent.com/Sharktheone/arch-mojo/main/install.py)
==> Making package: modular 0.2.0-1 (Thu 05 Oct 2023 09:32:24 PM +03)
==> Checking runtime dependencies...
==> Installing missing dependencies...
[sudo] password for hyousef:          
resolving dependencies...
looking for conflicting packages...

Packages (2) python-wheel-0.40.0-3  python-pip-23.2.1-1

Total Download Size:    3.21 MiB
Total Installed Size:  20.95 MiB

:: Proceed with installation? [Y/n] y
:: Retrieving packages...
:: Running post-transaction hooks...
==> Retrieving sources...
  -> Downloading modular-0.2.0-amd64.deb...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 45.7M  100 45.7M    0     0  5386k      0  0:00:08  0:00:08 --:--:-- 7664k

Packages (1) modular-0.2.0-1

Total Installed Size:  45.74 MiB

:: Proceed with installation? [Y/n] y
:: Running post-transaction hooks...
(1/2) Arming ConditionNeedsUpdate...
(2/2) Updating executables in /usr/bin...

Please enter your Modular auth token: xxxxxxxxxxxxxxxxxxxx

[44961:44961:20231005,213318.361332:ERROR filesystem_posix.cc:63] mkdir /home/hyousef/.modular/crashdb: No such file or directory (2)
[44963:44963:20231005,213318.362883:ERROR filesystem_posix.cc:63] mkdir /home/hyousef/.modular/crashdb: No such file or directory (2)
[44961:44961:20231005,213318.363103:ERROR socket.cc:120] recvmsg: Connection reset by peer (104)
Failed to initialize Crashpad.  Crash reporting will not be available.  Cause: crashpad failed to start handler
x - debian-binary
x - control.tar.xz
x - data.tar.xz
x - debian-binary
x - control.tar.xz
x - data.tar.xz
# Found release for https://packages.modular.com/mojo @ 0.3.1
# Installing to /home/hyousef/.modular/pkg/packages.modular.com_mojo
# Downloading artifacts. Please wait...

^C
Please restart your shell or run `source /home/hyousef/.zshrc `to complete the installation

~ source /home/hyousef/.zshrc
~ modular
modular: error: no command provided

usage: modular <command> [options]

For more information, try 'modular --help'.

~ modular install mojo
# Found release for https://packages.modular.com/mojo @ 0.3.1
# Installing to /home/hyousef/.modular/pkg/packages.modular.com_mojo
# Downloading artifacts. Please wait...
modular: error: failure detected - please run `modular clean` and try again
==========================
Failure Information:
	- Creating symlink for /home/hyousef/.modular/pkg/packages.modular.com_mojo/lib/libMojoJupyter.so.18git -> /home/hyousef/.modular/pkg/packages.modular.com_mojo/lib/libMojoJupyter.so failed: File exists
==========================

~ modular clean
~ modular install mojo
# Found release for https://packages.modular.com/mojo @ 0.3.1
# Installing to /home/hyousef/.modular/pkg/packages.modular.com_mojo
# Downloading artifacts. Please wait...
# Downloads complete, setting configs...
# Configs complete, running post-install hooks...
Installing collected packages:
Successfully installed ansiwrap-0.8.4 asttokens-2.4.0 attrs-23.1.0 backcall-0.2.0 certifi-2023.7.22 charset-normalizer-3.3.0 click-8.1.7 comm-0.1.4 debugpy-1.8.0 decorator-5.1.1 entrypoints-0.4 executing-2.0.0 fastjsonschema-2.18.1 find_libpython-0.3.0 idna-3.4 ipykernel-6.25.2 ipython-8.16.1 jedi-0.19.1 jsonschema-4.19.1 jsonschema-specifications-2023.7.1 jupyter-core-5.3.2 jupyter_client-8.3.1 matplotlib-inline-0.1.6 nbclient-0.8.0 nbformat-5.9.2 nest-asyncio-1.5.8 packaging-23.2 papermill-2.4.0 parso-0.8.3 pexpect-4.8.0 pickleshare-0.7.5 platformdirs-3.11.0 prompt-toolkit-3.0.39 psutil-5.9.5 ptyprocess-0.7.0 pure-eval-0.2.2 pygments-2.16.1 python-dateutil-2.8.2 pyyaml-6.0.1 pyzmq-25.1.1 referencing-0.30.2 requests-2.31.0 rpds-py-0.10.4 six-1.16.0 stack-data-0.6.3 tenacity-8.2.3 textwrap3-0.9.2 tornado-6.3.3 tqdm-4.66.1 traitlets-5.11.2 urllib3-2.0.6 wcwidth-0.2.8

[mojo] Testing: `mojo repl` [FAIL]
[mojo] Some components may have been installed successfully, but others may not work as expected. Please submit an issue to https://github.com/modularml/mojo and include the full output of the command you just ran.

🔥 Mojo installed! 🔥

Mojo's Python virtual environment created at /home/hyousef/.modular/pkg/packages.modular.com_mojo/venv

Now run the following commands if you are using bash:

echo 'export MODULAR_HOME="/home/hyousef/.modular"' >> ~/.bashrc
echo 'export PATH="/home/hyousef/.modular/pkg/packages.modular.com_mojo/bin:$PATH"' >> ~/.bashrc
source ~/.bashrc

If you are using ZSH, run the following commands:

echo 'export MODULAR_HOME="/home/hyousef/.modular"' >> ~/.zshrc
echo 'export PATH="/home/hyousef/.modular/pkg/packages.modular.com_mojo/bin:$PATH"' >> ~/.zshrc
source ~/.zshrc

Then enter 'mojo' to start the Mojo REPL.

For tool help, enter 'mojo --help'.
For more docs, see https://docs.modular.com/mojo.

~ echo $0
zsh
~ echo 'export MODULAR_HOME="/home/hyousef/.modular"' >> ~/.zshrc
~ echo 'export PATH="/home/hyousef/.modular/pkg/packages.modular.com_mojo/bin:$PATH"' >> ~/.zshrc
~ source ~/.zshrc

 ~ mojo --version
mojo 0.3.1 (a3eed7c8)
  ~ 
```

![image](https://github.com/Sharktheone/arch-mojo/assets/98168280/87cdf34f-7dd4-4a4b-9e02-c9eab16204e5)

![image](https://github.com/Sharktheone/arch-mojo/assets/98168280/190179c9-b951-476b-9520-98f92f932381)

**Comment by Sharktheone** [comment]
Yes, the crashpad thing is weird. This is since the update to v0.3.1. I'm not shure, but i guess this is due the missing `crashdb`directory in `~/.modular`. You can try creating it.

**Comment by hajsf** [comment]
> You can try creating it

I tried to create it now, but found it had been created, may be this is done thorough the `modular clean` process. thanks a lot for the support and the creating of this installer. 

![image](https://github.com/Sharktheone/arch-mojo/assets/98168280/adfab21a-9d0a-450a-a8ba-657e383990d5)

**Comment by hajsf** [comment]
I just want to confirm that I re-installed fresh copy of Archcraft, first thing I did is upgrading the existing packages, then mojo had been installed smoothly, I then installed VS code, and everything is fine there including Jupyter, thanks for your efforts:
 
![image](https://github.com/Sharktheone/arch-mojo/assets/98168280/ad1ce492-524b-4136-8a9e-9266fcdd6032)

![image](https://github.com/Sharktheone/arch-mojo/assets/98168280/0841cecf-90b2-4397-a176-75f220c45d09)

![image](https://github.com/Sharktheone/arch-mojo/assets/98168280/284c0dc4-feec-46a6-9f84-a1643dd71e7f)

![image](https://github.com/Sharktheone/arch-mojo/assets/98168280/00097577-c276-4e32-8706-f232c24e7200)

![image](https://github.com/Sharktheone/arch-mojo/assets/98168280/c7e52ca2-1eec-4165-95de-4e7ec17ffdbf)

![image](https://github.com/Sharktheone/arch-mojo/assets/98168280/9b899d66-3ae3-4722-8e6d-8afb8dd4c411)

![image](https://github.com/Sharktheone/arch-mojo/assets/98168280/adf445c5-0f4b-4810-a97b-d9fe8d0dc43c)

![image](https://github.com/Sharktheone/arch-mojo/assets/98168280/6dcbed6c-b173-4e49-8d72-33980ead2049)

**Comment by justsankalp** [comment]
> @mdmrk i have created a new issue from this, because i think, this is not related to this issue.

I still get this error, libpanel.so.6 is not present.. Any idea why?

**Comment by Sharktheone** [comment]
You mean when mojo tests it's repl? That's sadly a bit normal on the first installation. Modular spawns a new subprocess, which doesn't know of any of the libraries, with which modular got run.

However sometimes it works on my machine, even when I reset it to "default", I haven't got behind the logic of that.

**Comment by justsankalp** [comment]
> You mean when mojo tests it's repl? That's sadly a bit normal on the first installation. Modular spawns a new subprocess, which doesn't know of any of the libraries, with which modular got run.
> 
> However sometimes it works on my machine, even when I reset it to "default", I haven't got behind the logic of that.

Everything's fine now. Got it to run Roberta with Max.
